lib/Room.php

```diff
@@ -633,7 +634,20 @@ public function changeInCall($sessionId, $active) {
 	 * @return bool
 	 */
 	public function verifyPassword($password) {
-		return !$this->hasPassword() || $this->hasher->verify($password, $this->password);
+		$event = new GenericEvent($this, [
+			'password' => $password
+		]);
+		$this->dispatcher->dispatch(self::class . '::verifyPassword', $event);
+		if ($event->hasArgument('result')) {
+			$result = $event->getArgument('result');
+			return $result;
```

Comment:
I would prefer:
```php
return [
	'result' => $result['result'] ?? false,
	'url' => $result['url'] ?? ''
];
```
here, just to make sure that the values we require in our code are returned